apps/src/StudioApp.js

```diff
@@ -633,45 +636,44 @@ StudioApp.prototype.initVersionHistoryUI = function (config) {
 };
 
 StudioApp.prototype.getFirstContainedLevelResult_ = function () {
-  var results = this.getContainedLevelResults();
+  const results = codeStudioLevels.getContainedLevelResults();
   if (results.length !== 1) {
     throw "Exactly one contained level result is currently required.";
   }
   return results[0];
 };
 
 StudioApp.prototype.hasValidContainedLevelResult_ = function () {
-  var firstResult = this.getFirstContainedLevelResult_();
+  const firstResult = this.getFirstContainedLevelResult_();
   return firstResult.result.valid;
 };
 
 /**
  * @param {!AppOptionsConfig} config
  */
- StudioApp.prototype.notifyInitialRenderComplete = function (config) {
-  if (config.hasContainedLevels && config.containedLevelOps) {
-    this.lockContainedLevelAnswers = config.containedLevelOps.lockAnswers;
-    this.getContainedLevelResults = config.containedLevelOps.getResults;
-
+ StudioApp.prototype.notifyInitialRenderComplete = function () {
+  // TODO - better way to do this might be to have a component for containedLevelContainer
+  // that handles this in a lifecycle method (onComponentDidMount)
+  if (this.hasContainedLevels) {
```

Comment:
Even if I dont end up going the containedLevelContainer component approach, it seems likely that some of this logic is better placed either in codeStudioLevels, or in a separate module that takes care of contained levels stuff, so that studioApp mostly doesn't need to think about contained levels.